src/IonBinaryWriter.ts

```diff
@@ -96,6 +96,7 @@ export class BinaryWriter extends AbstractWriter {
     }
 
     writeBlob(value: Uint8Array): void {
+        _assertDefined(value);
         this.checkWriteValue();
         if (value === null || value === undefined) {
```

Comment:
There are quite a few places where the new call to `_assertDefined(value)` happens before an 

```typescript
if (value === null || value === undefined) {
//...
}
```

The above can now be simplified to `if (value === null)`.
```

Reply:
indeed, simplified